components/SearchInput.tsx

```diff
@@ -0,0 +1,60 @@
+import { Icon, Input, InputGroup, InputRightElement } from '@chakra-ui/react'
+import { HiOutlineSearch } from '@react-icons/all-files/hi/HiOutlineSearch'
+import { HiOutlineX } from '@react-icons/all-files/hi/HiOutlineX'
+import { useRouter } from 'next/router'
+import { FC, HTMLAttributes, useCallback, useMemo } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { removeEmptyFromObject } from '../utils'
+
+type IProps = HTMLAttributes<any> & {
```

Comment:
We could add directly the results of the `register` function in props so we could let the parent component do the `...register` as you describe for consistency and also to remove some logic from this component

```suggestion
type IProps = HTMLAttributes<any> & ReactFormRegisterReturnData {
```